web/packages/shared/components/AccessRequests/NewRequest/matcher.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * Teleport
+ * Copyright (C) 2024 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { AccessRequest, Resource } from 'shared/services/accessRequests';
+
+export function requestMatcher(
```

Comment:
I tried to rewrite it to make it more safe, but I ultimately couldn't find a good solution. It'd have been trivial if `requestMatcher` received the whole `AccessRequest` object rather than the value of `propName`:

```typescript
export function requestMatcher(
  accessRequest: AccessRequest,
  searchValue: string,
  propName: keyof AccessRequest
) {
  if (propName === 'roles') {
    return accessRequest[propName].some(role =>
      role.toUpperCase().includes(searchValue)
    );
  }

  if (propName === 'resources') {
    return accessRequest[propName].some(r =>
      Object.values(r.id)
        .concat(Object.values(r.details.hostname || {}))
        .concat(Object.values(r.details.friendlyName || {}))
        .some(v => v.toUpperCase().includes(searchValue))
    );
  }
}
```